backend/src/Edna/Config/Environment.hs

```diff
@@ -0,0 +1,86 @@
+module Edna.Config.Environment
+  ( EdnaEnvironments(..)
+
+  , ednaEnvs
+  , askDebugDB
+  ) where
+
+import Prelude (read)
+import Universum
+
+import Data.ByteString.Char8 (pack)
+import qualified Data.Char as C
+import Edna.Config.Definition (LoggingConfig(..))
+import Edna.Util (ConnString(..), NetworkAddress)
+import System.Environment (lookupEnv)
+
+-- If @EDNA_DB_DEBUG@ environment variable is set to @1@
+-- or @TRUE@ (case-insensitive), we will run postgres with
+-- debug logging.
+askDebugDB :: IO Bool
+askDebugDB = lookupEnv "EDNA_DB_DEBUG" <&> \case
+  Just "1"                       -> True
+  Just (map C.toLower -> "true") -> True
+  _                              -> False
+
+data EdnaEnvironments = EdnaEnvironments
+  { eeListen :: IO (Maybe NetworkAddress)
+  , eeDocs :: IO (Maybe Bool)
+  , eeConnection :: IO (Maybe ConnString)
+  , eeMaxCon :: IO (Maybe Int)
+  , eeLogging :: IO (Maybe LoggingConfig)
+  } deriving stock (Generic)
+
+ednaEnvs :: EdnaEnvironments
+ednaEnvs = EdnaEnvironments
+  { eeListen = apiListenEnv
+  , eeDocs = serveDocsEnv
+  , eeConnection = connectionEnv
+  , eeMaxCon = dbMaxConnectionsEnv
+  , eeLogging = loggingLevelEnv
+  }
+
+apiListenEnv :: IO (Maybe NetworkAddress)
+apiListenEnv = do
+  env <- lookupEnv "EDNA_API_LISTEN"
+  return $ read <$> env
```

Comment:
`read` is not good because it fails using `error` on bad input. We use `error` only for programmer mistakes, errors that are supposed to never happen no matter what input is provided. If `read` fails, the error may be triggered at some unexpected moment when `NetworkAddress` will be forced. There is `readEither` in `Universum`, it's better to use it here and throw some exception here when it fails (for example you can define `EnvParseError` type or use `throwString` from `safe-exceptions`).

Reply:
do you think here is better to throw exception or just put some notice message in IO and assume env is not set?

Reply:
I think it's better to throw an exception because "some notice message" is easier to accidentally miss. So if someone sets `EDNA_API_LISTEN` to something, I think they certainly intend to set listen address to something and if it doesn't work they want to restart properly anyway, so it's better to stop the program because it will be stopped anyway.

Reply:
agree